lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp

```diff
@@ -138,6 +138,129 @@ int pdu_session_resource_setup_request(
   return RETURNok;
 }
 
+int pdu_session_resource_release_request_retransmit(
+    ue_m5gmm_context_s* ue_context, amf_ue_ngap_id_t amf_ue_ngap_id,
+    smf_context_t* smf_ctx) {
+  bstring buffer;
+  uint32_t bytes                = 0;
+  DLNASTransportMsg* encode_msg = NULL;
+  SmfMsg* smf_msg               = NULL;
+  uint32_t len                  = 0;
+  uint32_t container_len        = 0;
+  amf_nas_message_t msg;
+  MessageDef* message_p = NULL;
+  int rc                = RETURNok;
+
+  memset(&msg, 0, sizeof(amf_nas_message_t));
+
+  msg.security_protected.plain.amf.header.extended_protocol_discriminator =
+      M5G_MOBILITY_MANAGEMENT_MESSAGES;
+  msg.security_protected.plain.amf.header.message_type = DLNASTRANSPORT;
+  msg.header.security_header_type = SECURITY_HEADER_TYPE_INTEGRITY_PROTECTED;
+  msg.header.extended_protocol_discriminator = M5G_MOBILITY_MANAGEMENT_MESSAGES;
+  msg.header.sequence_number =
+      ue_context->amf_context._security.dl_count.seq_num;
+
+  encode_msg = &msg.security_protected.plain.amf.msg.downlinknas5gtransport;
+  smf_msg    = &encode_msg->payload_container.smf_msg;
+
+  // NAS5g AmfHeader
+  encode_msg->extended_protocol_discriminator.extended_proto_discriminator =
+      M5G_MOBILITY_MANAGEMENT_MESSAGES;
+  len++;
+  encode_msg->spare_half_octet.spare  = 0x00;
+  encode_msg->sec_header_type.sec_hdr = 0x00;
+  len++;
+  encode_msg->message_type.msg_type = DLNASTRANSPORT;
+  len++;
+  encode_msg->payload_container.iei           = PAYLOAD_CONTAINER;
+  encode_msg->payload_container_type.iei      = PAYLOAD_CONTAINER_TYPE;
+  encode_msg->payload_container_type.type_val = N1_SM_INFO;
+  len++;
+  encode_msg->pdu_session_identity.iei = 0x12;
+  len++;
+  encode_msg->pdu_session_identity.pdu_session_id =
+      smf_ctx->smf_proc_data.pdu_session_identity.pdu_session_id;
+  len++;
+
+  // NAS SmfMsg
+  smf_msg->header.extended_protocol_discriminator =
+      M5G_SESSION_MANAGEMENT_MESSAGES;
+  smf_msg->header.pdu_session_id =
+      smf_ctx->smf_proc_data.pdu_session_identity.pdu_session_id;
+  OAILOG_INFO(
+      LOG_AMF_APP, "pdu_id:%d \n",
+      smf_ctx->smf_proc_data.pdu_session_identity.pdu_session_id);
```

Comment:
Just a general question, don't know if we can agree on this, could we log information with subscriber identity or other identity (ngap Id) if identy unknown ?